api/v1/authentication_types.go

```diff
@@ -100,8 +100,30 @@ type AuthenticationOIDC struct {
 	// +optional
 	// +kubebuilder:validation:Enum=Verify;InsecureSkip
 	EmailVerification *EmailVerificationType `json:"emailVerification,omitempty"`
+
+	// PromptTypes is an optional list of string values that specifies whether the identity provider prompts the end user
+	// for re-authentication and consent. See the RFC for more information on prompt types:
+	// https://openid.net/specs/openid-connect-core-1_0.html. The defined values are:
+	// ["None", "Login", "Consent", "SelectAccount"].
```

Comment:
Do we need to have explicit validation on PromptTypes? I can't remember if we automatically get validation that only the strings matching a PromptType will be allowed.
If not then I think a a user could specify anything and it would be passed through.

Reply:
Adding the kubebuilder magic to PromptType fixed it, thanks for pointing that out!
```
spec.oidc.promptTypes: Unsupported value: "oanasd": supported values: "None", "Login", "Consent", "SelectAccount"
```